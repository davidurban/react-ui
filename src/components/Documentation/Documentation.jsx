import PropTypes from 'prop-types';
import React from 'react';

const Documentation = props => (
  <div>
    <h3>{props.name}</h3>
    <div>
      {props.component}
    </div>
  </div>
);

Documentation.propTypes = {
  component: PropTypes.element.isRequired,
  name: PropTypes.string.isRequired,
};

export default Documentation;
