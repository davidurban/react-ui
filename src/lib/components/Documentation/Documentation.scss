@import '../../core/settings/colors';
@import '../../core/tools/spacing';

.root {
  padding: spacing(1);
  margin: spacing(1);
  background-color: map-get($colors-grays, gray-100);
}

.title {
  font-size: 1.25rem;
}

.sandbox {
  padding: 1.5rem;
  border: 2px dashed map-get($colors-grays, gray-400);
  background-color: $color-white;
}
