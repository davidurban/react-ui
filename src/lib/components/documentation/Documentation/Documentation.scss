@import '../../../styles/settings/colors';
@import '../../../styles/settings/typography';
@import '../../../styles/tools/offset';

.root {
  margin-bottom: offset(7);
}

.sandbox {
  padding: offset(5);
  border: 2px dashed map-get($colors-grays, gray-200);
  background-color: $color-white;
}
