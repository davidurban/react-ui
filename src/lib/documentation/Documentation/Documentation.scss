@import '../../shared/settings/colors';
@import '../../shared/settings/typography';
@import '../../shared/tools/spacing';

.root {
  margin-bottom: spacing(2);
}

.sandbox {
  padding: spacing(1);
  border: 2px dashed map-get($colors-grays, gray-200);
  background-color: $color-white;
}
