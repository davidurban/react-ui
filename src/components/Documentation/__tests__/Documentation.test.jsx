import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Documentation from '../index';

describe('rendering', () => {
  it('renders correctly mandatory props only', () => {
    const tree = shallow(<Documentation
      component={<div>Element</div>}
      name="Element"
    />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
