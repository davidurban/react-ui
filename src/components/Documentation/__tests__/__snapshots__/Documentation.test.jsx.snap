// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly mandatory props only 1`] = `
<div>
  <h3>
    Element
  </h3>
  <div>
    <div>
      Element
    </div>
  </div>
</div>
`;
